From: Chris Hofstaedtler <zeha@debian.org>
Date: Fri, 15 Mar 2024 18:13:07 +0100
Subject: Disable some problematic tests

--- /dev/null
+++ ruby3.1-3.1.2/test/excludes/TestFileUtils.rb
@@ -0,0 +1 @@
+exclude :test_chown, "fails on Launchpad builders"
--- /dev/null
+++ ruby3.1-3.1.2/test/excludes/TestProcess.rb
@@ -0,0 +1 @@
+exclude :test_execopts_gid, "fails on Launchpad builders"
--- /dev/null	2024-03-15 17:16:31.668000000 +0100
+++ ruby3.1-3.1.2/test/excludes/TestAddressResolve.rb	2024-03-15 18:12:06.000000000 +0100
@@ -0,0 +1 @@
+exclude :test_socket_getnameinfo_domain_blocking, "No internet in buildd" 
