From: Sorah Fukumori <sorah@cookpad.com>
Date: Fri, 26 Oct 2018 21:57:31 +0900
Subject: exclude several tests unstable in Docker

---
 test/dtrace/helper.rb | 1 +
 1 file changed, 1 insertion(+)

diff --git a/test/dtrace/helper.rb b/test/dtrace/helper.rb
index 023d116..500da5d 100644
--- a/test/dtrace/helper.rb
+++ b/test/dtrace/helper.rb
@@ -53,6 +53,7 @@ else
   cmd = false
 end
 ok &= system(*cmd, err: IO::NULL, out: IO::NULL) if cmd
+ok = false # Doesn't work properly inside Docker build
 
 module DTrace
   class TestCase < Test::Unit::TestCase
