From: Lucas Kanashiro <kanashiro@debian.org>
Date: Tue, 3 Oct 2023 18:41:55 -0300
Subject: Fix FTBS on hurd

Closes: #896509

Signed-off-by: Antonio Terceiro <terceiro@debian.org>
Signed-off-by: Samuel Thibault <samuel.thibault@ens-lyon.org>
---
 io.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/io.c b/io.c
index ca2cb90..581ac5e 100644
--- a/io.c
+++ b/io.c
@@ -2212,7 +2212,11 @@ io_writev(int argc, const VALUE *argv, VALUE io)
 
     for (i = 0; i < argc; i += cnt) {
 #ifdef HAVE_WRITEV
-        if ((fptr->mode & (FMODE_SYNC|FMODE_TTY)) && iovcnt_ok(cnt = argc - i)) {
+        if ((fptr->mode & (FMODE_SYNC|FMODE_TTY))
+# ifdef IOV_MAX
+		&& ((cnt = argc - i) < IOV_MAX)
+# endif
+	   	) {
             n = io_fwritev(cnt, &argv[i], fptr);
         }
         else
