From: Lucas Kanashiro <kanashiro@debian.org>
Date: Fri, 1 Nov 2019 15:41:55 -0300
Subject: Fix FTBS on hurd

Closes: #896509

Signed-off-by: Antonio Terceiro <terceiro@debian.org>
Signed-off-by: Samuel Thibault <samuel.thibault@ens-lyon.org>
---
 io.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/io.c
+++ b/io.c
@@ -1896,7 +1896,11 @@
 
     for (i = 0; i < argc; i += cnt) {
 #ifdef HAVE_WRITEV
-	if ((fptr->mode & (FMODE_SYNC|FMODE_TTY)) && iovcnt_ok(cnt = argc - i)) {
+        if ((fptr->mode & (FMODE_SYNC|FMODE_TTY))
+# ifdef IOV_MAX
+               && ((cnt = argc - i) < IOV_MAX)
+# endif
+               ) {
 	    n = io_fwritev(cnt, &argv[i], fptr);
 	}
 	else
