From: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date: Sat, 9 Jun 2018 10:17:47 -0300
Subject: Fix FTBFS on hurd

Closes: #896509

Signed-off-by: Antonio Terceiro <terceiro@debian.org>
Signed-off-by: Samuel Thibault <samuel.thibault@ens-lyon.org>
---
 io.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/io.c b/io.c
index 6192899..095060c 100644
--- a/io.c
+++ b/io.c
@@ -1646,7 +1646,11 @@ io_writev(int argc, VALUE *argv, VALUE io)
 
     for (i = 0; i < argc; i += cnt) {
 #ifdef HAVE_WRITEV
-	if ((fptr->mode & (FMODE_SYNC|FMODE_TTY)) && ((cnt = argc - i) < IOV_MAX)) {
+	if ((fptr->mode & (FMODE_SYNC|FMODE_TTY))
+# ifdef IOV_MAX
+		&& ((cnt = argc - i) < IOV_MAX)
+# endif
+		) {
 	    n = io_fwritev(cnt, &argv[i], fptr);
 	}
 	else
