Index: b/test/excludes/TestFileUtils.rb
===================================================================
--- /dev/null
+++ b/test/excludes/TestFileUtils.rb
@@ -0,0 +1 @@
+exclude :test_chown, "fails on Launchpad builders"
Index: b/test/excludes/TestProcess.rb
===================================================================
--- /dev/null
+++ b/test/excludes/TestProcess.rb
@@ -0,0 +1 @@
+exclude :test_execopts_gid, "fails on Launchpad builders"
