From: =?utf-8?q?C=C3=A9dric_Boutillier?= <boutil@debian.org>
Date: Tue, 6 Oct 2020 00:57:31 +0200
Subject: don't try to download stuff in configure

---
 defs/gmake.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/defs/gmake.mk b/defs/gmake.mk
index 7e9566f..8faa6c5 100644
--- a/defs/gmake.mk
+++ b/defs/gmake.mk
@@ -250,7 +250,7 @@ extract-gems: update-gems
 
 BUNDLED_GEMS := $(shell sed '/^[ 	]*\#/d;/^[ 	]*$$/d;s/[ 	][ 	]*/-/;s/[ 	].*//' $(srcdir)/gems/bundled_gems)
 
-update-gems: | $(patsubst %,gems/%.gem,$(BUNDLED_GEMS))
+update-gems:
 
 test-bundler-precheck: | $(srcdir)/.bundle/cache
 
