From: =?utf-8?q?C=C3=A9dric_Boutillier?= <boutil@debian.org>
Date: Tue, 6 Oct 2020 00:57:31 +0200
Subject: don't try to download stuff in configure

Reviewed-By: Lucas Kanashiro <kanashiro@debian.org>
Last-Update: 2023-10-03

--- a/defs/gmake.mk
+++ b/defs/gmake.mk
@@ -295,9 +295,7 @@
 bundled-gem-srcdir = $(srcdir)/gems/src/$(1)
 bundled-gem-extracted = $(srcdir)/.bundle/gems/$(1)-$(2)
 
-update-gems: | $(patsubst %,$(srcdir)/gems/%.gem,$(bundled-gems))
-update-gems: | $(call foreach-bundled-gems-rev,bundled-gem-gemfile)
-update-gems: | $(call foreach-bundled-gems-rev,bundled-gem-srcdir)
+update-gems:
 
 test-bundler-precheck: | $(srcdir)/.bundle/cache
 
