From: =?utf-8?q?C=C3=A9dric_Boutillier?= <boutil@debian.org>
Date: Tue, 6 Oct 2020 00:57:31 +0200
Subject: don't try to download stuff in configure

---
 defs/gmake.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/defs/gmake.mk b/defs/gmake.mk
index f093fd4..82df5d2 100644
--- a/defs/gmake.mk
+++ b/defs/gmake.mk
@@ -250,7 +250,7 @@ extract-gems: $(HAVE_BASERUBY:yes=update-gems)
 
 bundled-gems := $(shell sed '/^[ 	]*\#/d;/^[ 	]*$$/d;s/[ 	][ 	]*/-/;s/[ 	].*//' $(srcdir)/gems/bundled_gems)
 
-update-gems: | $(patsubst %,gems/%.gem,$(bundled-gems))
+update-gems:
 
 test-bundler-precheck: | $(srcdir)/.bundle/cache
 
