From: Sorah Fukumori <sorah@cookpad.com>
Date: Fri, 26 Oct 2018 21:57:31 +0900
Subject: exclude several tests unstable in Docker

---
 test/dtrace/helper.rb | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/test/dtrace/helper.rb b/test/dtrace/helper.rb
index ce73080..3ac1aa6 100644
--- a/test/dtrace/helper.rb
+++ b/test/dtrace/helper.rb
@@ -34,6 +34,10 @@ if ok
   end
 end
 
+
+ok = false # doesn't work properly inside Docker build
+
+
 # use miniruby to reduce the amount of trace data we don't care about
 rubybin = "miniruby#{RbConfig::CONFIG["EXEEXT"]}"
 rubybin = File.join(File.dirname(EnvUtil.rubybin), rubybin)
