From: Takashi Kokubun <takashikkbn@gmail.com>
Date: Thu, 4 Jan 2024 14:53:44 -0800
Subject: YJIT: Let RubyVM::YJIT.enable respect --yjit-stats

---
 yjit.rbinc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/yjit.rbinc b/yjit.rbinc
index 7acfeb1..6f417f6 100644
--- a/yjit.rbinc
+++ b/yjit.rbinc
@@ -3,6 +3,7 @@
 // auto-generated file
 //   by ./tool/mk_builtin_loader.rb
 //   with yjit.rb
+// + empty update from debian/patches to avoid regeneration attempt on debuild
 #include "internal/compilers.h"     /* for MAYBE_UNUSED */
 #include "internal/warnings.h"      /* for COMPILER_WARNING_PUSH */
 #include "ruby/ruby.h"              /* for VALUE */
